library/src/main/java/com/bumptech/glide/load/resource/gif/GifDrawable.java

```diff
@@ -390,6 +407,50 @@ public void setLoopCount(int loopCount) {
     }
   }
 
+  /**
+   * Register callback to listen to GifDrawable animation end event after specific loop count
+   * set by {@link GifDrawable#setLoopCount(int)}
+   *
+   * Note: Unregister this in onLoadCleared to avoid potential memory leak.
+   * @see GifDrawable#unregisterAnimationCallback(AnimationCallback).
+   *
+   * @param animationCallback Animation callback {@link Animatable2Compat.AnimationCallback}.
+   */
+  @Override
+  public void registerAnimationCallback(@NonNull AnimationCallback animationCallback) {
+    if (animationCallback == null) {
+      return;
+    }
+    if (animationCallbacks == null) {
+      animationCallbacks = new ArrayList<>();
+    }
+    animationCallbacks.add(animationCallback);
+  }
+
+  /**
+   * Unregister animation callback.
+   *
+   * @param animationCallback Animation callback {@link Animatable2Compat.AnimationCallback}.
+   * @return true when the animation callback is successfully removed, otherwise false.
+   */
+  @Override
+  public boolean unregisterAnimationCallback(@NonNull AnimationCallback animationCallback) {
+    if (animationCallbacks == null || animationCallback == null) {
```

Comment:
ditto here for animationCallback, I think you can skip the null check with the annotation (or make it a Precondition if you want one).